src/sourmash/signature.py

```diff
@@ -192,6 +192,16 @@ def __reduce__(self):
             ),
         )
 
+    def __copy__(self):
+        mh=self.minhash
+        a = SourmashSignature(
+            mh,
```

Comment:
great! hot take, tho, I think you should do `mh.to_frozen()` here which will make a new copy of the minhash. This would prevent copies of the signature from sharing mutable minhashes underneath, and in turn prevent this kind of code:

```
# ss is a SourmashSignature
tt = ss.copy()
tt.minhash.add_hash(5)

# I think people would not expect the 5 hash to be added to `ss.minhash` here.
assert ss != tt
```

could you do this and add a test checking that the above code works?

Reply:
I'm not sure I follow.

Reply:
Should I do this in the `__copy__` function itself?

Reply:
what are the options for implementing it?

Reply:
I'm not sure. Should I do this in the `__copy__` function, in addition to the things already in there, or should I do it instead of them?

Reply:
suggest writing the test, then figuring out the best way to implement it. that is sometimes how I do things :).